app/src/main/java/net/kwatts/powtools/App.java

```diff
@@ -21,4 +31,38 @@ public SharedPreferencesUtil getSharedPreferences() {
         }
         return sharedPreferencesUtil;
     }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        initWakeLock();
+        initDatabase();
+    }
+
+    private void initDatabase() {
+        db = Room.databaseBuilder(getApplicationContext(),
+                Database.class, "database-name-pow")
+                .fallbackToDestructiveMigration()
+                .build();
+        dbExecutor = Executors.newSingleThreadExecutor();
+    }
+
+    public static void dbExecute(DBExecutor dbExecutor) {
+        INSTANCE.dbExecutor.execute( () -> dbExecutor.run(INSTANCE.db));
+    }
+
+    private void initWakeLock() {
+        PowerManager powerManager = (PowerManager) getSystemService(POWER_SERVICE);
+        assert powerManager != null;
```

Comment:
An `assert` should only be used to hide an IDE warning when you are certain that a variable will never be `null`. Here `assert` is simply hiding the IDE warning and not offering any safety.

i.e. a crash will occur with or without the `assert` if `powerManager` is `null` but at least without `assert` you have an IDE warning to remind you that it should be fixed at some point

_Not your code originally, but being that your changes carried over the line I figured you kind of took ownership of it._ 😝